components/automate-chef-io/content/docs/iam-v2-api-reference.md

```diff
@@ -74,227 +159,556 @@ You can see the actions contained in the `editor` role in the [Roles section]({{
+
+### List Teams for a User
+
+The output lists all teams that contain the specified user.
+
+### Updating Your Own User
+
+You need admin permissions to update other users, but all local users have permission to update their own user details with this endpoint.
+For this endpoint, the `password` property is optional, so you only need to supply it if you wish to change your password.
+However, if you do supply `password` you must also supply one more property, `previous_password` as a security measure.
+
+*In the browser:* **Profile Menu**  >> **Profile**
+
+## Teams
+
+Method | HTTP request              | Description
+-------|---------------------------|------------
+list   | GET /teams               | lists all teams
+get    | GET /teams/***id***      | gets the specified team
+create | POST /teams              | creates a team
+update | PUT /teams/***id***      | updates a team
+delete | DELETE /teams/***id***   | deletes a team
+
+**Example Team:**
+
+```json
+{
+  "name": "team 1",
+  "id": "team-1",
+  "projects": [
+    "east-region",
+    "west-region"
+  ]
+}
+```
+
+### Listing Teams
+
+The output lists all local teams.
+
+*In the browser:* **Settings**  >> **Teams**
+
+### Getting a Team
+
+This shows the details of a single team, selected by its ID.
+
+*In the browser:* **Settings**  >> **Teams** >> [select a team]
+
+### Creating a Team
+
+Create a team by composing JSON with all the necessary team properties (see the beginning of the section above).
+
+*In the browser:* **Settings**  >> **Teams** >> Create Team
+
+### Updating a Team
+
+The update operation modifies *all* team properties, not just the ones you specify.
+Properties that you do not specify are reset to empty values.
+Thus, you should supply all of a team's properties, not just the ones you wish to update.
+Note that the ID is immutable; it can only be set at creation time.
+
+*In the browser:* **Settings**  >> **Teams** >> [select a team] >> Details
+
+### Deleting a Team
+
+Deleting a team is permanent and cannot be undone.
+
+*In the browser:* **Settings**  >> **Teams** >> [open control menu for target team] >> Delete Team
+
+## Team Membership
+
+A local team consists exclusively of local users.
+Note that the `user_ids` of the local users used here are GUIDs, generated when you create users.
+A user's GUID is reported in the `membership_id` property when you fetch a user.
+
+Method | HTTP request                           | Description
+-------|----------------------------------------|------------
+list   | GET /teams/***id***/users              | lists the membership
+add    | POST /teams/***id***/users:add         | add to the membership
+remove | POST /teams/***id***/users:remove      | remove from the membership
+
+**Example payload:**
+
+```json
+{
+  "id": "team-1",
+  "user_ids": [
+    "1a5a63d7-0b14-465a-aec1-5eea08a72343",
+    "6c4e3e8a-fade-4a9c-ac9f-99b86677530b"
+  ]
```

Comment:
It seems peculiar that for team membership we list users by GUIDs whereas for policy membership users (and teams and tokens, of course) are listed by id. 🤷‍♂

Reply:
That _is_ odd

Reply:
We use the `ID` to add members, though? That's kind of a problem given our lack of patchy updates . . .

Reply:
No, the GUID is used to reference members for input and for output, so it is consistent by itself, just not consistent with policy members.

Reply:
To make this consistent we'd probably need a table in the teams db that correlated v2 user IDs and v1 user GUIDs